src/hyperlight_host/src/sandbox/host_funcs.rs

```diff
@@ -181,20 +182,7 @@ fn call_host_func_impl(
             seccompiler::apply_filter(&seccomp_filter)?;
         }
 
-        #[cfg(feature = "function_call_metrics")]
-        {
-            let start = std::time::Instant::now();
-            let result = func.call(args.clone());
-            crate::histogram_vec_observe!(
-                &crate::sandbox::metrics::SandboxMetric::HostFunctionCallsDurationMicroseconds,
-                &[name],
-                start.elapsed().as_micros() as f64
-            );
-            result
-        }
-
-        #[cfg(not(feature = "function_call_metrics"))]
-        func.call(args)
+        HistogramMetric::time_and_emit_host_call(name.to_string(), || func.call(args.clone()))
```

Comment:
Again I am not sure its worth doing this this way, we need to call .to_string and args.clone each time we make a call which is overhead when not recording metrics

Reply:
Whoops fixed